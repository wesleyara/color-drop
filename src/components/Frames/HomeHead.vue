<template>
  <span class="flex flex-col items-center justify-center gap-2">
    <h1 class="mt-16 text-5xl">Color Drop</h1>

    <p class="text-xl text-center">
      A simple color dropper to search for products with similar shades!
    </p>
  </span>
</template>
