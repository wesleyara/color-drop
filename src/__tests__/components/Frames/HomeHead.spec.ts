/* eslint-disable @typescript-eslint/no-explicit-any */
import { describe, it, expect, beforeEach } from "vitest";
import { mount, VueWrapper } from "@vue/test-utils";
import HomeHead from "../../../components/Frames/HomeHead.vue";

describe("HomeHead unit tests", async () => {
  let sut: VueWrapper<any>;
  const titleMock = "Color Drop";

  beforeEach(async () => {
    sut = mount(HomeHead);
  });

  it("should render the component", () => {
    const wrapper = sut.exists();

    expect(wrapper).toBeDefined();
  });

  it("should match the snapshot", () => {
    const wrapper = sut.html();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render the component with the correct title", () => {
    const wrapper = sut.text();

    expect(wrapper).toContain(titleMock);
  });

  it("should have a class", () => {
    const wrapper = sut.classes();

    expect(wrapper).toContain("flex");
  });
});
