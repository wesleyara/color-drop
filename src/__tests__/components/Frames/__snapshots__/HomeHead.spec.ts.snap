// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeHead unit tests > should match the snapshot 1`] = `"<span class="flex flex-col items-center justify-center gap-2"><h1 class="mt-16 text-5xl">Color Drop</h1><p class="text-xl text-center"> A simple color dropper to search for products with similar shades! </p></span>"`;
